src/org/zaproxy/zap/extension/replacer/ReplacerParam.java

```diff
@@ -0,0 +1,230 @@
+        } catch (ConversionException e) {
+            logger.error("Error while loading the confirm remove rule option: " + e.getMessage(), e);
+        }
+
+        try {
+            this.showAdvancedOptions = getConfig().getBoolean(SHOW_ADV_OPTIONS_KEY, false);
+        } catch (ConversionException e) {
+            logger.error("Error while loading the show adv options option: " + e.getMessage(), e);
+        }
+    }
+
+    public List<ReplacerParamRule> getRules() {
+        return rules;
+    }
+
+    public void setRules(List<ReplacerParamRule> rules) {
+        this.rules = new ArrayList<>(rules);
+
+        ((HierarchicalConfiguration) getConfig()).clearTree(ALL_RULES_KEY);
+
+        ArrayList<String> enabledTokens = new ArrayList<>(rules.size());
+        for (int i = 0, size = rules.size(); i < size; ++i) {
+            String elementBaseKey = ALL_RULES_KEY + "(" + i + ").";
+            ReplacerParamRule rule = rules.get(i);
+
+            getConfig().setProperty(elementBaseKey + RULE_DESCRIPTION_KEY, rule.getDescription());
+            getConfig().setProperty(elementBaseKey + RULE_ENABLED_KEY, Boolean.valueOf(rule.isEnabled()));
+            getConfig().setProperty(elementBaseKey + RULE_MATCH_TYPE_KEY, rule.getMatchType().name());
+            getConfig().setProperty(elementBaseKey + RULE_MATCH_STRING_KEY, rule.getMatchString());
+            getConfig().setProperty(elementBaseKey + RULE_REGEX_KEY, Boolean.valueOf(rule.isMatchRegex()));
+            getConfig().setProperty(elementBaseKey + RULE_REPLACEMENT_KEY, rule.getReplacement());
+            List<Integer> initiators = rule.getInitiators();
+            if (initiators == null) {
+                getConfig().setProperty(elementBaseKey + RULE_INITIATORS_KEY, "");
+            } else {
+                getConfig().setProperty(elementBaseKey + RULE_INITIATORS_KEY, initiators.toString());
+            }
+
+            if (rule.isEnabled()) {
+                enabledTokens.add(rule.getDescription());
+            }
+        }
+
+        enabledTokens.trimToSize();
+        this.enabledRuleNames = enabledTokens;
+    }
+
+    public void addRule(String desc) {
+        this.rules.add(new ReplacerParamRule(desc, null));
+
+        this.enabledRuleNames.add(desc);
+    }
+
+    public void removeRule(String desc) {
+        this.rules.remove(new ReplacerParamRule(desc, null));
+
+        this.enabledRuleNames.remove(desc);
+    }
+    
+    public List<String> getEnabledRuleNames() {
+        return enabledRuleNames;
+    }
+
+    @ZapApiIgnore
+    public boolean isConfirmRemoveToken() {
+        return this.confirmRemoveToken;
+    }
+
+    @ZapApiIgnore
+    public void setConfirmRemoveToken(boolean confirmRemove) {
+        this.confirmRemoveToken = confirmRemove;
+        getConfig().setProperty(CONFIRM_REMOVE_RULE_KEY, Boolean.valueOf(confirmRemoveToken));
+    }
+
+    public boolean isShowAdvancedOptions() {

```

Comment:
This should be per rule (based on the value of `getInitiators()`) instead of a global setting?